File: python/3053.py
n = int(input())
resp = input()
for i in range(n):
    events = int(input())
    if events == 1:
        if resp == 'A':
            resp = 'B'
        elif resp == 'B':
            resp = 'A'
        elif resp == 'C':
            resp = 'C'
    elif events == 2:
        if resp == 'A':
            resp = 'A'
        elif resp == 'B':
            resp = 'C'
        elif resp == 'C':
            resp = 'B'
    elif events == 3:
        if resp == 'A':
            resp = 'C'
        elif resp == 'B':
            resp = 'B'
        elif resp == 'C':
            resp = 'A'

print(resp)

File: c++/3053.cpp
#include <bits/stdc++.h>
#include <iostream>

using namespace std;

int main() {
  int n;
  string resp;
  cin >> n >> resp;
  int events;

  for (int i = 0; i < n; i++) {
    cin >> events;
    if (events == 1 ) {
      if(resp == "A"){
        resp = "B";
      }else if(resp == "B"){
        resp = "A";
      }else if(resp == "C"){
        resp = "C";
      }
    } else if (events == 2) {
      if(resp == "A"){
        resp = "A";
      }else if(resp == "B"){
        resp = "C";
      }else if(resp == "C"){
        resp = "B";
      }
    } else if (events == 3) {
      if(resp == "A"){
        resp = "C";
      }else if(resp == "B"){
        resp = "B";
      }else if(resp == "C"){
        resp = "A";
      }
    }
  }
  cout << resp << endl;

  return 0;
}
